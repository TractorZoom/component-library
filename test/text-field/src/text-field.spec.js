import React from 'react';
import TextField from '../../../packages/text-field/src';
import renderer from 'react-test-renderer';

describe('text field', () => {
    it('should render with default margin and variant', () => {
        // given
        const props = {
            label: 'test',
            onChange: jest.fn(),
        };

        // when
        const component = renderer.create(<TextField {...props} />);
        const tree = component.toJSON();

        // then
        expect(tree).toMatchSnapshot();
    });

    it('should render with custom margin and variant', () => {
        // given
        const props = {
            label: 'test',
            margin: 'normal',
            onChange: jest.fn(),
            variant: 'filled',
        };

        // when
        const component = renderer.create(<TextField {...props} />);
        const tree = component.toJSON();

        // then
        expect(tree).toMatchSnapshot();
    });

    it('should render when type is date', () => {
        // given
        const props = {
            label: 'test',
            margin: 'normal',
            onChange: jest.fn(),
            type: 'date',
            variant: 'filled',
        };

        // when
        const component = renderer.create(<TextField {...props} />);
        const tree = component.toJSON();

        // then
        expect(tree).toMatchSnapshot();
    });
});
