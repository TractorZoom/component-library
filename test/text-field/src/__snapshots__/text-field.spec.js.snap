// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text field should render when type is date 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled"
    data-shrink={true}
  >
    test
  </label>
  <div
    className="MuiInputBase-root MuiFilledInput-root makeStyles-input-6 MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiFilledInput-input"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="date"
    />
  </div>
</div>
`;

exports[`text field should render when type is number 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
    data-shrink={false}
  >
    test
  </label>
  <div
    className="MuiInputBase-root MuiFilledInput-root makeStyles-input-6 MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiFilledInput-input"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      step={1}
      type="number"
    />
  </div>
</div>
`;

exports[`text field should render when type is number with custom step 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
    data-shrink={false}
  >
    test
  </label>
  <div
    className="MuiInputBase-root MuiFilledInput-root makeStyles-input-6 MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiFilledInput-input"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      step={1000}
      type="number"
    />
  </div>
</div>
`;

exports[`text field should render with custom margin and variant 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
    data-shrink={false}
  >
    test
  </label>
  <div
    className="MuiInputBase-root MuiFilledInput-root makeStyles-input-6 MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiFilledInput-input"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </div>
</div>
`;

exports[`text field should render with default margin and variant 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
    data-shrink={false}
  >
    test
  </label>
  <div
    className="MuiInputBase-root MuiFilledInput-root makeStyles-input-6 MuiInputBase-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiFilledInput-input"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </div>
</div>
`;
